### PR TITLE
[executorch] Always generate CustomOpsNativeFunctions.h if custom_ops.yaml is present

### PR DESCRIPTION
**Pull request by larryliu0820**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #95084

To match the build system logic, enforce CustomOpsNativeFunctions.h to be generated if we have custom_ops.yaml, even if we don't select any custom ops.

Added unit test.

Differential Revision: [D43402718](https://our.internmc.facebook.com/intern/diff/D43402718)

### CODE REVIEW

**Comment by larryliu0820** [comment]
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D43402718).

File: torchgen/gen_executorch.py [hunk 1]
```diff
@@ -291,15 +291,28 @@ def gen_functions_declarations(
 def gen_headers(
     *,
     native_functions: Sequence[NativeFunction],
+    gen_custom_ops_header: bool,
```
Comment (iseeyuan) [inline torchgen/gen_executorch.py:294]: What's the overhead if we always generate CustomOpsNativeFunctions.h?

**Review by iseeyuan — APPROVED** [review]
LGTM. If we could simplify to always generate CustomOpsNativeFunctions.h without much overhead it's even better.

File: torchgen/gen_executorch.py (hunk 1 above)
Comment (larryliu0820) [reply torchgen/gen_executorch.py:294]: Yeah I didn’t want to generate it if there’s no custom ops. Let me land this to unblock internal use case and then follow up.

**Comment by larryliu0820** [comment]
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D43402718).

**Pull request closed without merging**